modules/nlu/src/backend/election/ambiguous.ts

```diff
@@ -0,0 +1,21 @@
+import * as sdk from 'botpress/sdk'
+import _ from 'lodash'
+import { allInRange } from '../tools/math'
+import { NONE_INTENT } from './typings'
+
+export function detectAmbiguity(input: sdk.IO.EventUnderstanding): sdk.IO.EventUnderstanding {
+  // +- 10% away from perfect median leads to ambiguity
+  const totalConfInTopN = _.sumBy(input.intents, i => i.confidence)
+  const preds = input.intents.map(i => ({ ...i, confidence: i.confidence / totalConfInTopN }))
```

Comment:
this logic has changed a little:

I bring back the topN (which is a top2) on a scale of confidence where the sum is 100%. This was not necessary before, but it is now as the new election logic doesnt do this operation.